contracts/src/v0.8/dev/KeeperRegistryCheckUpkeepGasUsageWrapper.sol

```diff
@@ -0,0 +1,63 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.13;
+
+import {KeeperRegistryExecutableInterface} from "../KeeperRegistry.sol";
+import "../ConfirmedOwner.sol";
+
+/**
+ * @notice This contract serves as a wrapper around a keeper registry's checkUpkeep function.
+ */
+contract KeeperRegistryCheckUpkeepGasUsageWrapper is ConfirmedOwner {
+  KeeperRegistryExecutableInterface private immutable i_keeperRegistry;
+
+  /**
+   * @param keeperRegistry address of a keeper registry
+   */
+  constructor(KeeperRegistryExecutableInterface keeperRegistry) ConfirmedOwner(msg.sender) {
+    i_keeperRegistry = keeperRegistry;
+  }
+
+  /**
+   * @return the keeper registry
+   */
+  function getKeeperRegistry() external view returns (KeeperRegistryExecutableInterface) {
+    return i_keeperRegistry;
+  }
+
+  /**
+   * @notice This function is called by monitoring service to estimate how much gas checkUpkeep functions will consume.
+   * @param id identifier of the upkeep to check
+   */
+  function measureCheckGas(uint256 id)
```

Comment:
nit. can make this `cannotExecute` similar to checkUpkeep. It will ensure tx.origin is always 0 which is needed to call checkUpkeep

Reply:
Done